demos/sample_azure_iot_pnp/sample_azure_iot_pnp.c

```diff
@@ -0,0 +1,1062 @@
+ */
+#define sampleazureiotSUBSCRIBE_TIMEOUT                       ( 10 * 1000U )
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Unix time.
+ *
+ * @return Time in milliseconds.
+ */
+uint64_t ullGetUnixTime( void );
+/*-----------------------------------------------------------*/
+
+/* Define buffer for IoT Hub info.  */
+#ifdef democonfigENABLE_DPS_SAMPLE
+    static uint8_t ucSampleIotHubHostname[ 128 ];
+    static uint8_t ucSampleIotHubDeviceId[ 128 ];
+    static AzureIoTProvisioningClient_t xAzureIoTProvisioningClient;
+#endif /* democonfigENABLE_DPS_SAMPLE */
+
+static uint8_t ucScratchBuffer[ 128 ];
+
+/* Device values */
+static double xDeviceCurrentTemperature = sampleazureiotDEFAULT_START_TEMP_CELSIUS;
+static double xDeviceMaximumTemperature = sampleazureiotDEFAULT_START_TEMP_CELSIUS;
+static double xDeviceMinimumTemperature = sampleazureiotDEFAULT_START_TEMP_CELSIUS;
+static double xDeviceTemperatureSummation = sampleazureiotDEFAULT_START_TEMP_CELSIUS;
+static uint32_t ulDeviceTemperatureCount = sampleazureiotDEFAULT_START_TEMP_COUNT;
+static double xDeviceAverageTemperature = sampleazureiotDEFAULT_START_TEMP_CELSIUS;
+
+/* Command Values */
+static uint8_t ucCommandPayloadBuffer[ 256 ];
+static uint8_t ucCommandStartTimeValueBuffer[ 32 ];
+
+/* Property Values */
+static uint8_t ucPropertyPayloadBuffer[ 256 ];
+
+/* Each compilation unit must define the NetworkContext struct. */
+struct NetworkContext
+{
+    TlsTransportParams_t * pParams;
+};
+
+static AzureIoTHubClient_t xAzureIoTHubClient;
+/*-----------------------------------------------------------*/
+
+#ifdef democonfigENABLE_DPS_SAMPLE
+
+/**
+ * @brief Gets the IoT Hub endpoint and deviceId from Provisioning service.
+ *   This function will block for Provisioning service for result or return failure.
+ *
+ * @param[in] pXNetworkCredentials  Network credential used to connect to Provisioning service
+ * @param[out] ppucIothubHostname  Pointer to uint8_t* IoT Hub hostname return from Provisioning Service
+ * @param[in,out] pulIothubHostnameLength  Length of hostname
+ * @param[out] ppucIothubDeviceId  Pointer to uint8_t* deviceId return from Provisioning Service
+ * @param[in,out] pulIothubDeviceIdLength  Length of deviceId
+ */
+    static uint32_t prvIoTHubInfoGet( NetworkCredentials_t * pXNetworkCredentials,
+                                      uint8_t ** ppucIothubHostname,
+                                      uint32_t * pulIothubHostnameLength,
+                                      uint8_t ** ppucIothubDeviceId,
+                                      uint32_t * pulIothubDeviceIdLength );
+
+#endif /* democonfigENABLE_DPS_SAMPLE */
+
+/**
+ * @brief The task used to demonstrate the MQTT API.
+ *
+ * @param[in] pvParameters Parameters as passed at the time of task creation. Not
+ * used in this example.
+ */
+static void prvAzureDemoTask( void * pvParameters );
+
+/**
+ * @brief Connect to IoT Hub with reconnection retries.
```

Comment:
>IoT Hub with reconnection retries. [](http://example.com/codeflow?start=21&length=34)

connect to endpoint with retires

same function is used for dps connect. This fix is also required in other sample

Reply:
done in both